packages/components/src/Combobox/hooks/useCombobox.ts

```diff
@@ -14,18 +22,51 @@ type UseComboboxReturn = {
   selectedOptions: ComboboxOption[];
   selectedStateSetter: (selection: ComboboxOption[]) => void;
   shouldScroll: MutableRefObject<boolean>;
+  filteredOptions: ComboboxOption[];
 } & UseMakeComboboxHandlersReturn;
 
 export function useCombobox(
   selected: ComboboxOption[],
   onSelect: (selection: ComboboxOption[]) => void,
+  options: ComboboxOption[],
   onClose?: () => void,
   multiSelect?: boolean,
+  onSearchChange?: (searchValue: string) => void,
+  debounceTime: number = 300,
 ): UseComboboxReturn {
   const wrapperRef = useRef<HTMLDivElement>(null);
   const shouldScroll = useRef<boolean>(false);
   const [open, setOpen] = useState<boolean>(false);
   const [searchValue, setSearchValue] = useState<string>("");
+  const [filteredOptions, setFilteredOptions] =
```

Comment:
`filteredOptions` is a bit of a weird name. it fits really well when you are searching locally

but if you never search, or you provided an `onSearchChange` then it's possibly a misnomer. it might just be some options that aren't filtered, we can't really know - nor do we particularly care

Reply:
I don't think I knew it when I made this comment but my intuition was vaguely correct

this gets weird and causes a bug when you're not using the internal filtering